user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/MeasurementMain.java

```diff
@@ -345,8 +346,21 @@ private void downloadProteomicsMetadata() {
         context.experimentId().orElseThrow(() -> new ApplicationException(
             ErrorCode.GENERAL, null)),
         context.projectId().orElseThrow(() -> new ApplicationException(ErrorCode.GENERAL, null)));
+
+    Comparator<String> natOrder = Comparator.naturalOrder();
+
     var result = proteomicsMeasurements.stream().map(measurementPresenter::expandProteomicsPools)
-        .flatMap(Collection::stream).toList();
+        .flatMap(Collection::stream)
+        // sort by measurement codes first, then by sample codes
+        .sorted((e1, e2) -> {
+          int res = natOrder.compare(e1.measurementCode(), e2.measurementCode());
+          if(res!=0) {
+            return res;
+          }
+          String e1Code = e1.sampleInformation().sampleId();
+          String e2Code = e2.sampleInformation().sampleId();
+          return natOrder.compare(e1Code, e2Code);
+        }).toList();
```

Comment:
Since Java 8 you can use Comparator::comparing

```suggestion
        .flatMap(Collection::stream)
        .sorted(Comparator.comparing(ProteomicsMeasurementEntry::measurementCode)
            .thenComparing(e -> e.sampleInformation().sampleId()))
        .toList();
```

Reply:
nevermind, I think I found a solution :)